deerlab/bg_models.py

```diff
@@ -101,24 +112,61 @@ def _docstring(model,notes):
 .. math::
    D = \frac{\mu_0}{4\pi}\frac{(g_\mathrm{e}\mu_\mathrm{B})^2}{\hbar}
 """  
-def _hom3d_fcn(t,conc,lam):
-#---------------------------------------------------------------------------------------
+def _hom3d(t,conc,lam):
     # Units conversion    
     conc = conc*1e-6*1e3*Nav # umol/L -> mol/L -> mol/m^3 -> spins/m^3
     # Compute background function
-    B = np.exp(-8*pi**2/9/m.sqrt(3)*lam*conc*D*np.abs(t*1e-6))
+    κ = 8*pi**2/9/m.sqrt(3)
+    B = np.exp(-κ*lam*conc*D*np.abs(t*1e-6))
     return B
-#---------------------------------------------------------------------------------------
 # Create model
-bg_hom3d = Model(_hom3d_fcn,constants='t')
-bg_hom3d.description = 'Background from homogeneous distribution of spins in a 3D medium'
+bg_hom3d = Model(_hom3d,constants='t')
+bg_hom3d.description = 'Background from a homogeneous distribution of spins in a 3D medium'
 # Parameters
 bg_hom3d.conc.set(description='Spin concentration', lb=0.01, ub=5000, par0=50, units='μM')
 bg_hom3d.lam.set(description='Pathway amplitude', lb=0, ub=1, par0=1, units='')
 # Add documentation
 bg_hom3d.__doc__ = _docstring(bg_hom3d,notes)
 
 
+#=======================================================================================
+#                                     bg_hom3d_phase
+#=======================================================================================
+notes = r"""
+**Model**
+
+This model describes the phase shift due to inter-molecular interactions between one observer spin with a 3D homogenous distribution of spins of concentration `c_s`
+
+.. image:: ../images/model_scheme_bg_hom3d.png
+   :width: 350px
+
+The expression for this model is
+
+.. math::
+   B(t) = \mathrm{exp}\left(-\frac{8\pi^2}{9\sqrt{3}}\frac{\sqrt{3} + ln(2 - \sqrt{3})}{\pi}\lambda c_s D |t|\right)`
+
+where `c_s` is the spin concentration (entered in spins/m\ :sup:`3` into this expression) and D is the dipolar constant
+
+.. math::
+   D = \frac{\mu_0}{4\pi}\frac{(g_\mathrm{e}\mu_\mathrm{B})^2}{\hbar}
+"""  
+def _hom3dphase(t,conc,lam):
+    # Units conversion    
+    conc = conc*1e-6*1e3*Nav # umol/L -> mol/L -> mol/m^3 -> spins/m^3
+    # Compute background function
+    ξ = 8*pi**2/9/np.sqrt(3)*(np.sqrt(3)+np.log(2-np.sqrt(3)))/np.pi*D
+    B = np.exp(-1j*ξ*lam*conc*(t*1e-6))
```

Comment:
This doesn't look quite right, since there is no real part in the exponent.

Reply:
Because there is no real part. I separate the phase shift from the decay contribution in the models. 

For example: 
- `bg_hom3d` models the decay of the signal due to the spins in the environment (i.e. the real part)
- `bg_hom3d_phase` models the phase shift due to the spins in the environment  (i.e. the imaginary part)

The full complex-valued background function is `bg_hom3d*bg_hom3d_phase`. This avoids duplication of the code and model of the real part.

Reply:
I see